src/09_shader_modules.rs

```diff
@@ -0,0 +1,294 @@
+extern crate env_logger;
+#[cfg(feature = "dx12")]
+extern crate gfx_backend_dx12 as back;
+#[cfg(feature = "metal")]
+extern crate gfx_backend_metal as back;
+#[cfg(feature = "vulkan")]
+extern crate gfx_backend_vulkan as back;
+extern crate gfx_hal as hal;
+extern crate winit;
+#[macro_use]
+extern crate lazy_static;
+extern crate glsl_to_spirv;
+
+use hal::{Capability, Device, Instance, PhysicalDevice, QueueFamily, Surface, SwapchainConfig};
+use std::{fs, io::Read};
+
+static WINDOW_NAME: &str = "09_shader_modules";
+
+fn main() {
+    log_output_setup();
+    let (window, events_loop) = init_window();
+    // we must return surface, since it must be destroyed before swapchain
+    let (_instance, device, swapchain, _surface, frame_images) = init_hal(&window);
+    main_loop(events_loop);
+    // must destroy the swapchain
+    clean_up(device, frame_images, swapchain);
+}
+
+lazy_static! {
```

Comment:
I don't think we get much benefit from`lazy_static` here. Could we simply `include_str!("src/09_shader_base.vert")` and `compile` while we're creating the module instead?